src/transformers/models/patchtsmixer/modeling_patchtsmixer.py

```diff
@@ -0,0 +1,2265 @@
+            )
+
+        self.sequence_length = sequence_length
+        self.patch_length = patch_length
+        self.stride = stride
+
+        # get the number of patches
+        self.num_patches = compute_num_patches(sequence_length, patch_length, stride)
+        new_sequence_length = patch_length + stride * (self.num_patches - 1)
+        self.s_begin = sequence_length - new_sequence_length
+
+    def forward(self, past_values: torch.Tensor):
+        r"""
+        Args:
+            past_values (`torch.Tensor` of shape (batch_size x sequence_length x num_input_channels), *required*):
+                Context values of the time series.
+        Returns:
+            (`torch.Tensor` of shape (batch_size x num_input_channels x num_patches x patch_length)):
+                Output tensor data.
+        """
+        sequence_length = past_values.shape[-2]
+
+        if sequence_length != self.sequence_length:
+            raise ValueError(
+                f"Input sequence length ({sequence_length}) doesn't match model configuration ({self.sequence_length})."
+            )
+
+        x = past_values[:, self.s_begin :, :]  # x: [batch_size x new_sequence_length x num_channels]
+        x = x.unfold(
+            dimension=-2, size=self.patch_length, step=self.stride
+        )  # x: [batch_size x num_patches x num_input_channels x patch_length]
+        x = x.transpose(-2, -3).contiguous()  # x: [batch_size x num_input_channels x num_patches x patch_length]
+        return x
+
+
+# TODO: add copied from after PatchTST master merge
+class PatchTSMixerMasking(nn.Module):
+    """
+    Class to random or forcast masking.
+
+    Args:
+        mask_type (str, *optional*): Masking type. Allowed values are random, forecast. Defaults to random.
+        mask_ratio (`float`, *optional*): Mask ratio.
+        mask_patches (`list`, *optional*): List of patch lengths to mask in the end of the data.
+        mask_patch_ratios (`list`, *optional*): List of weights to use for each patch length. For Ex. if
+            patch_lengths is [5,4] and mix_ratio is [1,1], then equal weights to both patch lengths. Defaults to None.
+        unmasked_channel_indices (`list`, *optional*):
+            Control Variable channel indices. These channels will not be masked. Defaults to None.
+        channel_consistent_masking (bool, *optional*):
+            When true, masking will be same across all channels of a timeseries. Otherwise, masking positions will vary
+            across channels. Defaults to True.
+        mask_value (`int`, *optional*): Value to use for masking. Defaults to 0.
+        seed_number (`int`, *optional*): Random seed, when None seed is not set. Defaults to None.
+    """
+
+    def __init__(
+        self,
+        mask_type: str = "random",
+        mask_ratio=0.5,
+        mask_patches: list = [2, 3],
+        mask_patch_ratios: list = [1, 1],
+        channel_consistent_masking: bool = False,
+        unmasked_channel_indices: list = None,
```

Comment:
same here for whatever can be taken from the config

Reply:
corrected